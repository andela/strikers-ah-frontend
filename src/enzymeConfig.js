import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
