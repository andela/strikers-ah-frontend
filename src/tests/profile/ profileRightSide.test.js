/* eslint-disable no-undef */
import '../../enzymeConfig';
import RightProfile from '../../components/common/profileRightSide';
import { findByTestAttribute, getConnectedComponent } from './test-helpers';

describe('test profile right side', () => {
  describe('Render profile right side', () => {
    let wrapper;
    beforeEach(() => {
      const props = {
        accountInfo: {
          firstname: 'Mwibutsa',
          lastname: 'floribert',
          username: 'fmwibutsa',
          followers: [],
          followings: [],
        },
        editProfile: {
          toggleShowImage: { toggleForm: jest.fn() },
          showForm: false,
          toggleEditProfile: jest.fn(),
        },
        userArticles: [
          {
            id: 1,
            title: 'hello this is fake title',
            description: 'this is a dummy description',
          },
        ],
      };
      wrapper = getConnectedComponent(RightProfile, {}, props);
    });
    it('should render profile right side', () => {
      const profile = findByTestAttribute(wrapper, 'accountInfoComponent');
      expect(profile.length).toBe(1);
    });
  });
  describe('Render profile right side', () => {
    let wrapper;
    beforeEach(() => {
      const props = {
        accountInfo: {
          firstname: 'Mwibutsa',
          lastname: 'floribert',
          username: 'fmwibutsa',
          followers: [],
          followings: [],
          onwer: true,
        },
        editProfile: {
          toggleShowImage: { toggleForm: jest.fn() },
          showForm: false,
          toggleEditProfile: jest.fn(),
        },
        userArticles: [],
      };
      wrapper = getConnectedComponent(RightProfile, {}, props);
    });
    it('should render profile right side', () => {
      const profile = findByTestAttribute(wrapper, 'accountInfoComponent');
      expect(profile.length).toBe(1);
    });
  });
});
