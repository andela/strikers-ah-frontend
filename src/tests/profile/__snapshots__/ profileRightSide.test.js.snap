// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test profile right side Render profile right side  should render profile right side 1`] = `ShallowWrapper {}`;

exports[`test profile right side Render profile right side should render profile right side 1`] = `ShallowWrapper {}`;
